### ISSUE TITLE
Add support for Firestore DocumentReference

### ISSUE THREAD

**Issue by sejr** [sejr/firegraph#3]
Right now we assume all values are primitives, e.g. their actual value can be stripped and put into a JavaScript object. The `DocumentReference` can behave that way, but it would be easier to allow us to retrieve those references during resolution.

Example:

``` graphql
query {
    posts {
        id
        message
        author {
            id
            fullName
        }
    }
}
```

**Comment by sejr** [comment]
Because this is how we also handle searching collections (nested data between `{` and `}`), we might want to come up with special syntax to handle References. For example:

``` graphql
query {
    posts {
        id
        message

        # Nested Collection
        comments {
            id
            message
        }

        # Document Reference
        author(fromCollection: 'users') {
            id
            fullName
        }
    }
}
```

**Comment by sejr** [comment]
Support with raw referenced document IDs has landed in `firegraph@1.0.3`. We still need to handle Firestore's built-in DocumentReference type.

**Comment by Taosif7** [comment]
In a Firestore document, the reference type of fields have DocumentReference type of objects in data. Using this fact, detection of reference type fields can be done, and hence using their path, we can resolve document.
